gwpy/utils/tests/test_sphinx_zenodo.py

```diff
@@ -0,0 +1,157 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with GWpy.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for :mod:`gwpy.utils.sphinx.zenodo`.
+"""
+
+import pytest
+
+import requests
+
+from ...testing.errors import pytest_skip_network_error
+from ..sphinx import zenodo as gwpy_zenodo
+
+GITHUB_RELEASE_API_URL = "https://api.github.com/repos/gwpy/gwpy/releases"
+
+# simplified output of zenodo API query (as of 18/Oct/2023)
+MOCK_ZENODO_API_JSON = {
+    "hits": {
+        "hits": [
+            {
+                "conceptdoi": "10.5281/zenodo.597016",
+                "conceptrecid": "597016",
+                "created": "2023-10-05T10:29:34.092078+00:00",
+                "doi": "10.5281/zenodo.8409995",
+                "doi_url": "https://doi.org/10.5281/zenodo.8409995",
+                "files": None,
+                "id": 8409995,
+                "links": None,
+                "metadata": {
+                    "title": "gwpy/gwpy: GWpy 3.0.7",
+                    "version": "v3.0.7",
+                },
+                "title": "gwpy/gwpy: GWpy 3.0.7",
+            },
+            {
+                "conceptdoi": "10.5281/zenodo.597016",
+                "conceptrecid": "597016",
+                "created": "2023-10-05T09:51:21.482810+00:00",
+                "doi": "10.5281/zenodo.8409892",
+                "doi_url": "https://doi.org/10.5281/zenodo.8409892",
+                "files": None,
+                "links": None,
+                "metadata": {
+                    "title": "gwpy/gwpy: GWpy 3.0.6",
+                    "version": "v3.0.6",
+                },
+                "title": "gwpy/gwpy: GWpy 3.0.6",
+            },
+        ],
+    },
+}
+MOCK_ZENODO_API_RST = """
+-----
+3.0.7
+-----
+
+.. image:: https://zenodo.example.com/badge/doi/10.5281/zenodo.8409995.svg
+    :alt: gwpy/gwpy: GWpy 3.0.7 Zenodo DOI badge
+    :target: https://doi.org/10.5281/zenodo.8409995
+
+-----
+3.0.6
+-----
+
+.. image:: https://zenodo.example.com/badge/doi/10.5281/zenodo.8409892.svg
+    :alt: gwpy/gwpy: GWpy 3.0.6 Zenodo DOI badge
+    :target: https://doi.org/10.5281/zenodo.8409892
+""".strip()
+
+
+@pytest.fixture
+@pytest_skip_network_error
+def latest():
+    """Get the latest release of GWpy from the GitHub API.
+    """
+    resp = requests.get(
+        GITHUB_RELEASE_API_URL,
+        params={"per_page": 1},
+    )
+    resp.raise_for_status()
+    return resp.json()[0]
+
+
+@pytest_skip_network_error
+def test_zenodo_format_citations_latest(latest):
+    """Check that :func:`gwpy.utils.sphinx.zenodo.format_citations` includes
+    the latest actual release in the output.
+    """
+    rst = gwpy_zenodo.format_citations(
+        597016,
+    )
+    latestversion = latest["tag_name"].lstrip("v")
+    latesthead = "-" * len(latestversion)
+    assert f"""
+{latesthead}
+{latestversion}
+{latesthead}""".strip() in rst
```

Comment:
Is this formatting intentional?

Reply:
I'm not sure what you are asking. This `assert` checks that an RST-format header for the latest-discovered version is present in the rendered text, so 'yes' this is intentional.

Reply:
I support this could have been achieved via `'\n'.join` or something a little cleaner than a block quote, but heyho.

Reply:
Sorry, I meant the indentation.

Reply:
(but it's not a problem)